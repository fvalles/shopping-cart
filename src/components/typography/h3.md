Basic h3 text component with default color="productTitle":

```jsx
<H3>Lorem ipsum</H3>
```

H3 text component with color="primary":

```jsx
<H3 color="primary">Lorem ipsum</H3>
```
