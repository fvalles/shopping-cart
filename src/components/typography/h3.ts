import styled from 'styled-components';
import { TypographyProps } from './common';

export const H3 = styled.h3<TypographyProps>`
  color: ${({ color = 'productTitle', theme }) => theme.Colors[color]};
  font-family: ${({ theme }) => theme.Fonts.light}, 'Avenir Next', 'Segoe UI',
    sans-serif;
  font-size: 10px;
  letter-spacing: 1px;
  line-height: 16px;
`;
