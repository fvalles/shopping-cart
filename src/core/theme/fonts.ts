export const Fonts = {
  light: 'Avenir Light',
  medium: 'Avenir Medium',
  regular: 'Avenir Regular',
};

export type FontsType = typeof Fonts;
