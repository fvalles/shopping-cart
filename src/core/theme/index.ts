export { Colors } from './colors';
export { Fonts } from './fonts';
export { GlobalStyle } from './global-style';
