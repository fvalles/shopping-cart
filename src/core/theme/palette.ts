export const palette = {
  // color names are based on https://colornamer.netlify.app/
  black: '#000000',
  blackMarket: '#212240',
  blackMarketLight: '#21224029',
  bleachedSilk: '#F3F3F3',
  faintFern: '#DBDBE0',
  mischka: '#A6A7B3',
  sharpGrey: '#CACAD1',
  stormGrey: '#717285',
  venetianNights: '#7350FF',
  white: '#FFFFFF',
};
