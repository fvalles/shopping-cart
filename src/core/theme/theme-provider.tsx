import React, { FunctionComponent } from 'react';
import { DefaultTheme, ThemeProvider as Provider } from 'styled-components';
import { Colors, Fonts } from '.';

interface ThemeProviderProps {
  children?: React.ReactNode;
}

export const ThemeProvider: FunctionComponent<ThemeProviderProps> = ({
  children,
}) => (
  <Provider
    theme={
      {
        Colors,
        Fonts,
      } as DefaultTheme
    }>
    {children}
  </Provider>
);
