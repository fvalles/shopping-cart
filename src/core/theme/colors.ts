import { palette } from './palette';

export const Colors = {
  // color names are based on https://colornamer.netlify.app/
  primary: palette.venetianNights,
  productImageBorder: palette.sharpGrey,
  productPrice: palette.black,
  productQuantity: palette.faintFern,
  productTitle: palette.mischka,
  summaryBackground: palette.bleachedSilk,
  summaryHeading: palette.stormGrey,
  summaryText: palette.blackMarket,
  white: palette.white,
};

export type ColorsType = typeof Colors;
export type KeyColors = keyof ColorsType;
