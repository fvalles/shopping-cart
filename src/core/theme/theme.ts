import { DefaultTheme } from 'styled-components';
import { Colors } from './colors';

const theme: DefaultTheme = {
  Colors,
};

export { theme };
