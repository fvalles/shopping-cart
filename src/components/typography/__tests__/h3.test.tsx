import React from 'react';
import { ThemeProvider } from '@core/theme/theme-provider';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { H3 } from '../h3';

describe('h3 component', () => {
  it('should render with default css properties', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <H3>Lorem ipsum</H3>
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('color', '#A6A7B3');
    expect(result).toHaveStyleRule(
      'font-family',
      "Avenir Light,'Avenir Next','Segoe UI', sans-serif",
    );
    expect(result).toHaveStyleRule('font-size', '10px');
    expect(result).toHaveStyleRule('letter-spacing', '1px');
    expect(result).toHaveStyleRule('line-height', '16px');
    expect(result).toMatchSnapshot();
  });

  it('should render with custom white color', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <H3 color="white">Lorem ipsum</H3>
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('color', '#FFFFFF');
    expect(result).toHaveStyleRule(
      'font-family',
      "Avenir Light,'Avenir Next','Segoe UI', sans-serif",
    );
    expect(result).toHaveStyleRule('font-size', '10px');
    expect(result).toHaveStyleRule('letter-spacing', '1px');
    expect(result).toHaveStyleRule('line-height', '16px');
    expect(result).toMatchSnapshot();
  });
});
