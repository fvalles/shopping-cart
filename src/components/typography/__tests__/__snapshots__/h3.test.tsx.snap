// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`h3 component should render with custom white color 1`] = `
.c0 {
  color: #FFFFFF;
  font-family: Avenir Light,'Avenir Next','Segoe UI', sans-serif;
  font-size: 10px;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  line-height: 16px;
}

<h3
  className="c0"
  color="white"
>
  Lorem ipsum
</h3>
`;

exports[`h3 component should render with default css properties 1`] = `
.c0 {
  color: #A6A7B3;
  font-family: Avenir Light,'Avenir Next','Segoe UI', sans-serif;
  font-size: 10px;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  line-height: 16px;
}

<h3
  className="c0"
>
  Lorem ipsum
</h3>
`;
